src/connector.ts

```diff
@@ -106,14 +160,35 @@ class StargazerConnector extends AbstractConnector {
     this.emitDeactivate();
   }
 
+  async switchEVMProvider(chain: Chains) {
+    // Check if the evm chain is supported
+    if (!['ethereum', 'bsc', 'polygon', 'avalanche'].includes(chain)) {
+      throw new StargazerConnectorError('Unsupported chain');
+    }
+
+    // Deactivate the previous provider -> Remove all listeners
+    await this.deactivate();
+
+    // Set the new provider
+    const provider = await this.getChainProvider(chain);
+    this.#activeEVMProvider = provider;
+
+    // Activate the provider -> Add all listeners
+    await this.activate();
+
+    // Emit an update event on the chain id
+    const chainId = await this.getChainId();
+    this.onChainChanged(chainId);
+  }
+
```

Comment:
Who calls this? The implementing Dapp?

Reply:
Yeah. This method is called from the Web3-React demo site.